### PR TITLE
Compress Estimate struct

### PR DESCRIPTION
**Pull request by mibac138**
Size reduced from 72 bytes to 48 bytes.

### CODE REVIEW

File: src/utils.rs [hunk 1]
```diff
@@ -16,64 +16,77 @@ pub fn secs_to_duration(s: f64) -> Duration {
 }
 
 pub struct Estimate {
-    buf: Vec<f64>,
-    buf_cap: usize,
-    last_idx: usize,
+    buf: Box<[f64; 15]>,
```
Comment (djc) [inline src/utils.rs:19]: The mechanisms you're using here could use a bunch of comments -- it's not at all clear to me what the approach is. Also, do you have an actual use case where this matters or do you just like squeezing out performance? :)

Comment (mibac138) [reply src/utils.rs:20]: I do like squeezing out performance and that's the main reason, however there were a few issues related to performance and this is a small stepping stone in improving it.

File: src/utils.rs [hunk 2]
```diff
@@ -15,8 +15,17 @@ pub fn secs_to_duration(s: f64) -> Duration {
     Duration::new(secs, nanos)
 }
 
+/// Ring buffer with constant capacity
```
Comment (djc) [inline src/utils.rs:18]: Would you mind adding some notes about how/where this is used?

File: src/utils.rs (hunk 1 above)
Comment (djc) [reply src/utils.rs:22]: I'm curious what kind of scenarios you're seeing where issues like these impact performance. Can you say more about that?

**Comment by djc** [comment]
Thanks!

**Pull request merged**